mne/time_frequency/tfr.py

```diff
@@ -1107,9 +1162,246 @@ def plot(self, picks, baseline=None, mode='mean', tmin=None, tmax=None,
                         ylim=None, tfr=data[idx: idx + 1], freq=freqs,
                         x_label='Time (ms)', y_label='Frequency (Hz)',
                         colorbar=colorbar, cmap=cmap, yscale=yscale)
+
+            if title is 'auto':
+                if aggregate is None or len(info['ch_names']) == 1:
+                    title = info['ch_names'][0]
+                else:
+                    ch_title = ", ".join(info['ch_names'][:6])
```

Comment:
Can we refactor the plot_compare_evoked code that does the same thing and only use one private helper function? (`_format_ch_names`)